dashboard/modules/snapshot/snapshot_head.py

```diff
@@ -197,8 +218,34 @@ async def _get_job_activity_info(self, timeout: int) -> RayActivityResponse:
                     if job_table_entry.end_time
                     else latest_job_end_time
                 )
+
                 if not is_dead and not in_internal_namespace:
-                    num_active_drivers += 1
+                    if not use_pending_tasks or is_running_tasks:
+                        num_active_drivers += 1
+                    else:
+                        # For alive idle drivers, the latest job end time should be
+                        # the latest end time of its tasks.
+                        option = ListApiOptions(
+                            filters=[
+                                ("job_id", "=", job_table_entry.job_id.hex()),
+                            ],
+                            detail=True,
+                            timeout=timeout,
+                        )
+                        tasks = await self._state_api.list_tasks(option=option)
```

Comment:
using state-api here might not be the right choice. state-api is designed for observability use-cases and has a 10,000 task limit. We may miss some pending tasks. 

With observability apis, we may even turn it off for some clusters when the cluster is too large and its impacting the main workload.

I recommend checking with @rickyyx or @rkooo567 if there's a more reliable API to use here. It might actually be better for core to track the pending task liveness in GCS.

Reply:
What's the definition of "last active time of the driver"? 

If there's task currently running/pending, the last active time is now right?
If there's no tasks running, then the last active time is the latest time of finished/errored task. 

I feel a seperate endpoint to GCS directly that asks for this should be better. Or we also have the idle_time information of each node in GCS now, although it doesn't have an API exposed yet I believe.  The idle time information is derived from worker and resource usage of each node, which I believe is used by the autoscaler idle detection, will this be a better source of truth? Or am I understanding the requirements wrong.